eos-tech-support/eos-fix-ostree-repo

```diff
@@ -166,29 +175,55 @@ def mark_commits_partial(repo):
             print('Commit', checksum, 'already marked as partial')
             continue
 
+        mark_partial = False
         try:
-            repo.traverse_commit(checksum, 0)
+            # If a dirtree is missing, traverse_commit will fail with
+            # G_IO_ERROR_NOT_FOUND.
+            _, reachable_objects = repo.traverse_commit(checksum, 0)
+
+            # Unfortunately, it doesn't check that the leaves (dirmeta
+            # and files) exist, so we need to do that manually. In case
+            # that behavior ever changes, just check that all the
+            # reachable objects exist.
+            for commit_obj in reachable_objects:
+                if commit_obj in found_objects:
+                    # Already found this object
+                    continue
+
+                commit_obj_checksum, commit_obj_type = \
+                    OSTree.object_name_deserialize(commit_obj)
+
+                _, have_object = repo.has_object(commit_obj_type,
+                                                 commit_obj_checksum)
```

Comment:
Oh, I just realized I don't have to do this lookup at all. I already passed in the set of all objects in the repo in `all_objects`. I'm going to test that out now.